tests/test_queries/test_queryset_level_methods.py

```diff
@@ -76,6 +78,27 @@ class Meta:
     pairs: pydantic.Json = ormar.JSON(default=["2", "3"])
 
 
+class Customer(ormar.Model):
+    class Meta:
+        metadata = metadata
+        database = database
+        tablename = "customer"
+
+    class QuerySetCls(QuerySet):
+
+        async def first_or_404(self, *args, **kwargs):
+            entity = await self.get_or_none(*args, **kwargs)
+            if not entity:
+                # maybe HTTPException in fastapi
+                raise ValueError("customer not found")
+            return entity
+
+    __queryset_cls__ = QuerySetCls
```

Comment:
This should be added to docs in sections of models as a way to overwrite setting for main/default `QuerySet`